superset-frontend/src/dashboard/components/gridComponents/Tabs.jsx

```diff
@@ -33,7 +33,38 @@ import getLeafComponentIdFromPath from '../../util/getLeafComponentIdFromPath';
 import { componentShape } from '../../util/propShapes';
 import { NEW_TAB_ID, DASHBOARD_ROOT_ID } from '../../util/constants';
 import { RENDER_TAB, RENDER_TAB_CONTENT } from './Tab';
-import { TAB_TYPE } from '../../util/componentTypes';
+import { CHART_TYPE, TAB_TYPE } from '../../util/componentTypes';
+import { getChartIdsInFilterScope } from '../../util/activeDashboardFilters';
+
+const findTabsWithChartsInScope = (
+  dashboardLayout,
+  chartsInScope,
+  childId,
+  tabId,
+  tabsToHighlight,
+) => {
+  if (
+    dashboardLayout[childId].type === CHART_TYPE &&
+    chartsInScope.includes(dashboardLayout[childId].meta.chartId)
+  ) {
+    tabsToHighlight.add(tabId);
+  }
+  if (
+    dashboardLayout[childId].children.length === 0 ||
+    (dashboardLayout[childId].type === TAB_TYPE && tabsToHighlight.has(childId))
+  ) {
+    return;
+  }
+  dashboardLayout[childId].children.forEach(childId =>
```

Comment:
Just to avoid confusion, would it make sense to call the second `childId` something like `subChildId`?

Reply:
Nice catch, I haven't noticed that name clash. Fixed